### PR TITLE
VSCODE-79: move playground to the language server

### PR DESCRIPTION
**Pull request by alenakhineika**
## Description

- In this PR I added instantiation of Language Server to the mdbExtensionController and moved playground execution to this Language Server.

- The playground makes a call to the Language Server Client and sends two parameters: code that needs to be evaluated and driverUrl.

- The Language Server Client sends a request with these values to the Language Server Server.

- Looks like we can't use DataService in the Language Server Server. Probably because it extends from EventEmitter. But we can use CliServiceProvider from '@mongosh/service-provider-server' instead. It accepts 2 params driverUrl and driverOptions. It provides similar methods as DataService does: https://github.com/mongodb-js/mongosh/blob/master/packages/service-provider-server/src/cli-service-provider.ts. Please let me know if you have any objections? The investigation ticket was created: https://jira.mongodb.org/browse/VSCODE-81

- I have noticed that the "Extension + Server Inspector" and "Attach to Language Server" runners from Launch file do not work. The first thing it should be 6009 instead of 6005, but even if I change it, Vscode complains anyway that a connection is refused.

- Moved code from the `activate` method of the Language Server to the `constructor` in order to set `this.client` value in the constructor and get rid of later checks on undefined in each method with sendRequest.

### Checklist
- [x] New tests and/or benchmarks are included
- [ ] Documentation is changed or added

## Motivation and Context

- [ ] Bugfix
- [x] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [x] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by alenakhineika** [comment]
Hey @imlucas, after our whiteboarding meeting, I have realized, that I need to try it right away, instead of getting deeper into theory. It will help me to better understand the concept. This is how I see the moving of playground to the language server. And what do you think about using  CliServiceProvider instead of DataService? Looking forward to your feedback.

File: src/language/server.ts [hunk 1]
```diff
@@ -275,9 +277,17 @@ connection.onCompletionResolve(
...
-  // connection.console.log(`executeAll: ${JSON.stringify(event)}`);
-  return '';
+connection.onRequest('executeAll', async (params) => {
+  try {
+    const connectionOptions = params.connectionOptions || {};
+    const runtime = new ElectronRuntime(
+      await CliServiceProvider.connect(params.connectionString, connectionOptions)
+    );
+
+    return await runtime.evaluate(params.codeToEvaluate);
```
Comment (Anemy) [inline src/language/server.ts:287]: I think we could move this error into a new try that way a code error can be better handled. Currently if there's an error in a playground's code, like calling a method that doesn't exist the error message is `Unable to connect to MongoDB instance. Make sure it is started correctly.`. Does the runtime throw an error that we can use?

File: src/test/suite/editors/playgroundController.test.ts [hunk 2]
```diff
@@ -8,56 +11,169 @@ import {
...
+  const docPath = path.resolve(__dirname, '../../../../src/test/fixture', docName);
 
-const sinon = require('sinon');
+  return vscode.Uri.file(docPath);
+};
+
+/**
+ * Opens the MongoDB playground
+ */
+export async function openPlayground(docUri: vscode.Uri) {
```
[the remark this answers is not in the document]
Comment (alenakhineika) [reply src/test/suite/editors/playgroundController.test.ts:31]: Removing it!

**Comment by alenakhineika** [comment]
Created a ticket for cancellation token: https://jira.mongodb.org/browse/VSCODE-82

File: src/language/server.ts (hunk 1 above)
Comment (alenakhineika) [reply src/language/server.ts]: Removed try-catch at all, since they are being caught in the playground controller anyway.

**Pull request merged**